File: 41_3_Parliament/Python/IndianParliamentSystem.py
class President:
    def __init__(self, name, term_start_date):
        self.name = name
        self.term_start_date = term_start_date

    def display(self):
        print(f"President Name: {self.name}")
        print(f"Term Start Date: {self.term_start_date}")

class Member:
    def __init__(self, name, party, house):
        self.name = name
        self.party = party
        self.house = house

    def display(self):
        print(f"Name: {self.name}")
        print(f"Party: {self.party}")
        print(f"House: {self.house}")

class Bill:
    def __init__(self, title, description, status):
        self.title = title
        self.description = description
        self.status = status

    def display(self):
        print(f"Bill Title: {self.title}")
        print(f"Description: {self.description}")
        print(f"Status: {self.status}")

class Session:
    def __init__(self, session_name, start_date, end_date):
        self.session_name = session_name
        self.start_date = start_date
        self.end_date = end_date

    def display(self):
        print(f"Session Name: {self.session_name}")
        print(f"Start Date: {self.start_date}")
        print(f"End Date: {self.end_date}")

def main():
    president = None
    rajya_sabha_members = []
    lok_sabha_members = []
    bills = []
    sessions = []

    while True:
        print("\nIndian Parliament System Menu:")
        print("1. Set President")
        print("2. Add Member to Rajya Sabha")
        print("3. Add Member to Lok Sabha")
        print("4. Add Bill")
        print("5. Add Session")
        print("6. Display President")
        print("7. Display Rajya Sabha Members")
        print("8. Display Lok Sabha Members")
        print("9. Display Bills")
        print("10. Display Sessions")
        print("11. Exit")
        choice = int(input("Enter your choice: "))

        if choice == 1:
            name = input("Enter President's Name: ")
            term_start_date = input("Enter Term Start Date: ")
            president = President(name, term_start_date)
            print("President set successfully.")
        elif choice == 2:
            name = input("Enter Member's Name: ")
            party = input("Enter Party: ")
            rajya_sabha_members.append(Member(name, party, "Rajya Sabha"))
            print("Rajya Sabha member added successfully.")
        elif choice == 3:
            name = input("Enter Member's Name: ")
            party = input("Enter Party: ")
            lok_sabha_members.append(Member(name, party, "Lok Sabha"))
            print("Lok Sabha member added successfully.")
        elif choice == 4:
            title = input("Enter Bill Title: ")
            description = input("Enter Bill Description: ")
            status = input("Enter Bill Status: ")
            bills.append(Bill(title, description, status))
            print("Bill added successfully.")
        elif choice == 5:
            session_name = input("Enter Session Name: ")
            start_date = input("Enter Session Start Date: ")
            end_date = input("Enter Session End Date: ")
            sessions.append(Session(session_name, start_date, end_date))
            print("Session added successfully.")
        elif choice == 6:
            if president:
                president.display()
            else:
                print("President not set.")
        elif choice == 7:
            if not rajya_sabha_members:
                print("No Rajya Sabha members added.")
            else:
                for member in rajya_sabha_members:
                    member.display()
                    print()
        elif choice == 8:
            if not lok_sabha_members:
                print("No Lok Sabha members added.")
            else:
                for member in lok_sabha_members:
                    member.display()
                    print()
        elif choice == 9:
            if not bills:
                print("No bills added.")
            else:
                for bill in bills:
                    bill.display()
                    print()
        elif choice == 10:
            if not sessions:
                print("No sessions added.")
            else:
                for session in sessions:
                    session.display()
                    print()
        elif choice == 11:
            print("Exiting...")
            break
        else:
            print("Invalid choice. Please enter a number between 1 and 11.")

if __name__ == "__main__":
    main()

File: 41_3_Parliament/C++/IndianParliamentSystem.cpp
#include <iostream>
#include <vector>
#include <string>

using namespace std;

class President {
    string name;
    string termStartDate;
public:
    President(string name, string termStartDate) : name(name), termStartDate(termStartDate) {}
    void display() {
        cout << "President Name: " << name << endl;
        cout << "Term Start Date: " << termStartDate << endl;
    }
};

class Member {
    string name;
    string party;
    string house;
public:
    Member(string name, string party, string house) : name(name), party(party), house(house) {}
    void display() {
        cout << "Name: " << name << endl;
        cout << "Party: " << party << endl;
        cout << "House: " << house << endl;
    }
};

class Bill {
    string title;
    string description;
    string status;
public:
    Bill(string title, string description, string status) : title(title), description(description), status(status) {}
    void display() {
        cout << "Bill Title: " << title << endl;
        cout << "Description: " << description << endl;
        cout << "Status: " << status << endl;
    }
};

class Session {
    string sessionName;
    string startDate;
    string endDate;
public:
    Session(string sessionName, string startDate, string endDate) : sessionName(sessionName), startDate(startDate), endDate(endDate) {}
    void display() {
        cout << "Session Name: " << sessionName << endl;
        cout << "Start Date: " << startDate << endl;
        cout << "End Date: " << endDate << endl;
    }
};

int main() {
    President* president = nullptr;
    vector<Member> rajyaSabhaMembers;
    vector<Member> lokSabhaMembers;
    vector<Bill> bills;
    vector<Session> sessions;

    int choice;
    string name, party, title, description, status, sessionName, startDate, endDate;

    do {
        cout << "\nIndian Parliament System Menu:" << endl;
        cout << "1. Set President" << endl;
        cout << "2. Add Member to Rajya Sabha" << endl;
        cout << "3. Add Member to Lok Sabha" << endl;
        cout << "4. Add Bill" << endl;
        cout << "5. Add Session" << endl;
        cout << "6. Display President" << endl;
        cout << "7. Display Rajya Sabha Members" << endl;
        cout << "8. Display Lok Sabha Members" << endl;
        cout << "9. Display Bills" << endl;
        cout << "10. Display Sessions" << endl;
        cout << "11. Exit" << endl;
        cout << "Enter your choice: ";
        cin >> choice;
        cin.ignore(); // Consume newline

        switch (choice) {
            case 1:
                cout << "Enter President's Name: ";
                getline(cin, name);
                cout << "Enter Term Start Date: ";
                getline(cin, termStartDate);
                delete president; // Clean up previous instance
                president = new President(name, termStartDate);
                cout << "President set successfully." << endl;
                break;
            case 2:
                cout << "Enter Member's Name: ";
                getline(cin, name);
                cout << "Enter Party: ";
                getline(cin, party);
                rajyaSabhaMembers.push_back(Member(name, party, "Rajya Sabha"));
                cout << "Rajya Sabha member added successfully." << endl;
                break;
            case 3:
                cout << "Enter Member's Name: ";
                getline(cin, name);
                cout << "Enter Party: ";
                getline(cin, party);
                lokSabhaMembers.push_back(Member(name, party, "Lok Sabha"));
                cout << "Lok Sabha member added successfully." << endl;
                break;
            case 4:
                cout << "Enter Bill Title: ";
                getline(cin, title);
                cout << "Enter Bill Description: ";
                getline(cin, description);
                cout << "Enter Bill Status: ";
                getline(cin, status);
                bills.push_back(Bill(title, description, status));
                cout << "Bill added successfully." << endl;
                break;
            case 5:
                cout << "Enter Session Name: ";
                getline(cin, sessionName);
                cout << "Enter Session Start Date: ";
                getline(cin, startDate);
                cout << "Enter Session End Date: ";
                getline(cin, endDate);
                sessions.push_back(Session(sessionName, startDate, endDate));
                cout << "Session added successfully." << endl;
                break;
            case 6:
                if (president != nullptr) {
                    president->display();
                } else {
                    cout << "President not set." << endl;
                }
                break;
            case 7:
                if (rajyaSabhaMembers.empty()) {
                    cout << "No Rajya Sabha members added." << endl;
                } else {
                    for (auto& m : rajyaSabhaMembers) {
                        m.display();
                        cout << endl;
                    }
                }
                break;
            case 8:
                if (lokSabhaMembers.empty()) {
                    cout << "No Lok Sabha members added." << endl;
                } else {
                    for (auto& m : lokSabhaMembers) {
                        m.display();
                        cout << endl;
                    }
                }
                break;
            case 9:
                if (bills.empty()) {
                    cout << "No bills added." << endl;
                } else {
                    for (auto& b : bills) {
                        b.display();
                        cout << endl;
                    }
                }
                break;
            case 10:
                if (sessions.empty()) {
                    cout << "No sessions added." << endl;
                } else {
                    for (auto& s : sessions) {
                        s.display();
                        cout << endl;
                    }
                }
                break;
            case 11:
                cout << "Exiting..." << endl;
                break;
            default:
                cout << "Invalid choice. Please enter a number between 1 and 11." << endl;
        }
    } while (choice != 11);

    delete president; // Clean up before exiting
    return 0;
}
